cube/src/main/java/zone/cogni/asquare/cube/shacl/ShaclToConversionProfile.java

```diff
@@ -131,20 +131,40 @@ private CompactConversionProfile.Attribute calculateAttribute(@Nonnull CompactCo
     attribute.setProperty(calculateProperty(propertyShape));
     attribute.setSingle(calculateSingle(propertyShape));
     attribute.setType(calculateAttributeType(propertyShape));
-    attribute.setInverse(false);
+    attribute.setInverse(calculateInverse(propertyShape));
     return attribute;
   }
 
+  private boolean calculateInverse(Resource propertyShape) {
+    if (propertyShape.hasProperty(Shacl.path)) return false;
+    if (propertyShape.hasProperty(Shacl.inversePath)) return true;
+
+    throw new RuntimeException("missing 'path' or 'inversePath' on property '" + propertyShape.getURI() + "'");
```

Comment:
should also something special happen if both are true?

Reply:
yeah. should not happen. I will add a check.